server/webapp/WEB-INF/rails.new/app/assets/javascripts/version_updater.js

```diff
@@ -0,0 +1,87 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END**********************************/
+
+var VersionUpdater = function (staleVersionInfoUrl, updateServerVersionInfoUrl) {
+    this.staleVersionInfoUrl = staleVersionInfoUrl;
+    this.updateServerVersionInfoUrl = updateServerVersionInfoUrl;
+}
+
+jQuery.extend(VersionUpdater.prototype, {
+    update: function () {
+        if(!this.canUpdateVersion()){
+            return;
+        }
+        this.fetchStaleVersionInfo();
+    },
+
+    fetchStaleVersionInfo: function(){
+        _this = this;
+        jQuery.ajax({
```

Comment:
Referring to `jQuery` is a PITA. Using a javascript self invoking function makes this a bit easier –

```javascript
(function ($) {
  ...
  $.ajax()
  ...
})(jQuery);
```